### DISCUSSION TITLE
Compare between nx plugins

### DISCUSSION THREAD

**Discussion by jon9090** [nrwl/nx#17080] [General]
Nx has many plugins such: `@nx/js`, `@nx/node`, `@nx/esbuild`, `@nx/rspack`, `@nx/vite`, `@nx/webpack`.

Could someone can compare all those packages? when to use one over the other? when I want to create a lib in node should I use `@nx/node` or `@nx/js`? why are both not in one package? what is the different between application with rspack and node? using vite or esbuild I can create an application.

Can somebody explain those plugins? but in when to use when and why?

**Comment by unleashit** [comment]
> when I want to create a lib in node should I use @nx/node or @nx/js? why are both not in one package? 

I wonder the same thing. There's @nx/js:tsc for example which I assume depends on Node. 

Biggest thing missing from NX for me is better docs on the executors. How to use and combine for the desired results, ideally mapped to the end commands that actually get called.
